lib/stdio.hoon

```diff
@@ -133,6 +163,24 @@
 ::
 ::    ----
 ::
+::  Identity is immutable
+::
+++  get-identity
+  =/  m  (async ,@p)
+  ^-  form:m
+  |=  =async-input
+  [~ ~ ~ %done our.bowl.async-input]
+::
+::  Entropy is always increasing
```

Comment:
Is it? If I issue three async get-entropy calls in a row, do I get different entropy each time?

Reply:
I don't think so.  lib/async could make that happen by cycling the entropy on every `;<  ... bind:m` call.

Reply:
Right, it's just the per-event entropy (and the comment is just an attempt symmetry with `+get-time` and `+get-identity`). Entropy would have to become part of the `async` state in order to be cycled.

This is one case of several where the interaction between a tapp and the state of the `bowl` is subtle and potentially unexpected (since the tapp may evaluate a command in a different event than that which enqueued it). I'm tempted to say the tapp shouldn't have any direct access to the bowl, so as to be forced to use monadic readers that can be made correct and predictable.

Reply:
The changes I made to `contracts` are in the same vein. Usually, all the events in transaction take place on top of the same duct/bone (pushing and popping wires in a straightforward stack discipline), in which case a contract can just be canceled on the current bone. But the `+tapp-admin/*` generators violate this, and it's possible to imagine other scenarios that do as well, so I made the contracts always track the bone one which they were established.